hawkbit-autoconfigure/src/main/java/org/eclipse/hawkbit/autoconfigure/security/InMemoryUserManagementAutoConfiguration.java

```diff
@@ -93,13 +90,29 @@ UserDetailsService userDetailsService() {
         if (userPrincipals.isEmpty()) {
             final String name = securityProperties.getUser().getName();
             final String password = securityProperties.getUser().getPassword();
-            userPrincipals.add(new UserPrincipal(name, password, name, name, name, null, DEFAULT_TENANT,
-                    PermissionUtils.createAllAuthorityList()));
+            final List<String> roles = securityProperties.getUser().getRoles();
+            List<GrantedAuthority> authorityList;
+            if (roles.isEmpty()) {
+                authorityList = PermissionUtils.createAllAuthorityList();
+            } else {
+                authorityList = createAuthoritiesFromList(roles);
+            }
```

Comment:
for unary conditions like this, I prefer to use the conditional operator:
```
List<GrantedAuthority> authorityList = roles.isEmpty() ? PermissionUtils.createAllAuthorityList() : createAuthoritiesFromList(roles);
```
But that's of course a coding style and in no way a requirement ;)